db/migrate/20170315122946_generate_shortcodes_for_responses.rb

```diff
@@ -0,0 +1,30 @@
+class Response < ActiveRecord::Base
+  belongs_to :mission
+  belongs_to :form
+
+  def generate_shortcode
+    shortcode_chars = ("a".."z").to_a + ("0".."9").to_a
+
+    begin
+      response_code = 5.times.map { shortcode_chars.sample }.join
+      mission_code = mission.shortcode
+      form_code = form.code
+      shortcode = [mission_code, form_code, response_code].join("-")
+      puts shortcode
+      self.shortcode = shortcode
+    end while Response.exists?(shortcode: self.shortcode)
+  end
+end
+
+class GenerateShortcodesForResponses < ActiveRecord::Migration
+  def up
+    Response.find_each do |response|
```

Comment:
Perhaps eager load mission and form here to speed this up? I don't think you'd need to test it, just add `.includes(:form, :mission)`.